hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/bulk/BulkInsertWriterHelper.java

```diff
@@ -212,5 +212,17 @@ private static WriteStatus toWriteStatus(HoodieInternalWriteStatus internalWrite
     writeStatus.setTotalErrorRecords(internalWriteStatus.getTotalErrorRecords());
     return writeStatus;
   }
+
+  /**
+   * Get partition path from the record.
+   * @param record
+   * @return
+   */
+  public String getPartitionPath(RowData record) {
+    if (keyGen == null) {
+      return null;
+    }
```

Comment:
We can still fetch the partition path if the keygen is not null.

Reply:
yes, I will try to use some other way, like the RowDataKeyGen#getRecordPartitionPath